src/browser/vdiff.js

```diff
@@ -1,28 +1,106 @@
-import { chai, expect } from '@open-wc/testing';
+import { chai, expect, nextFrame } from '@open-wc/testing';
 import { executeServerCommand } from '@web/test-runner-commands';
 
-let test;
+let test, soonPromise;
 
-chai.Assertion.addMethod('golden', function(...args) {
-	return ScreenshotAndCompare.call({ test, elem: this._obj }, ...args); // eslint-disable-line no-invalid-this
+/* eslint-disable no-undef, no-invalid-this */
+chai.Assertion.addMethod('golden', async function(...args) {
+	await soonPromise?.catch(err => expect.fail(err));
+	return ScreenshotAndCompare.call({ test, elem: this._obj }, ...args);
 });
-mocha.setup({ // eslint-disable-line no-undef
+
+chai.Assertion.addChainableMethod('soon',
+	() => { throw new TypeError('"soon" is not a function'); },
+
+	async function() {
+		let resolve, reject;
+		soonPromise = new Promise((res, rej) => {
+			resolve = res;
+			reject = rej;
+		});
+
+		try {
+			expect(this._obj).to.be.instanceof(HTMLElement);
+		} catch (e) {
+			reject(e.message);
+		}
+
+		await nextFrame();
+		await this._obj.updateComplete;
+		resolve();
+	}
+);
+
+mocha.setup({
 	rootHooks: {
 		beforeEach() {
 			test = this.currentTest;
 		}
 	}
 });
+/* eslint-enable */
 
 async function ScreenshotAndCompare(opts) {
+	await document.fonts.ready; // firefox fonts
```

Comment:
For some reason Firefox fonts go back into `loading`, so we need to double check they are done before the screenshot.

Reply:
Yeah it should be.